src/Sylius/Component/Core/Translation/TranslatableEntityLocaleAssigner.php

```diff
@@ -21,14 +22,23 @@
 
 final class TranslatableEntityLocaleAssigner implements TranslatableEntityLocaleAssignerInterface
 {
-    public function __construct(private LocaleContextInterface $localeContext, private TranslationLocaleProviderInterface $translationLocaleProvider)
-    {
-    }
+    public function __construct(
+        private LocaleContextInterface $localeContext,
+        private TranslationLocaleProviderInterface $translationLocaleProvider,
+        private ?CommandBasedContextCheckerInterface $commandBasedChecker = null
+    ) { }
 
     public function assignLocale(TranslatableInterface $translatableEntity): void
     {
         $fallbackLocale = $this->translationLocaleProvider->getDefaultLocaleCode();
 
+        if ($this->commandBasedChecker !== null && $this->commandBasedChecker->isRunningFromCommand()) {
```

Comment:
Seeing this plus the extra constructor argument got me thinking, why not decorate this service instead of modifying it?

Reply:
I created 3rd not required argument along with deprecation note, i think both cases are good.